packages/client/src/components/AppNavBar/AppNavBar.tsx

```diff
@@ -5,199 +5,215 @@ import MenuIcon from '@mui/icons-material/Menu';
+			window.localStorage.removeItem('userRole');
+			// eslint-disable-next-line no-empty
+		} catch { }
+	};
+
+	const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+		setMenuAnchorElem(event.currentTarget);
+	};
+
+	const handleMenuClosed = () => {
+		setMenuAnchorElem(null);
+	};
+
+	const handleProfileMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+		setProfileMenuAnchorElem(event.currentTarget);
+	};
+
+	const handleProfileMenuClosed = () => {
+		setProfileMenuAnchorElem(null);
+	};
+
+	return (
+		<AppBar position='static'>
+			<Toolbar sx={{ width: 'min(100%, 92rem)', mx: 'auto' }}>
+				<Box sx={{ display: { xs: 'block', md: 'none' } }}>
+					<IconButton size='large' onClick={handleMenuOpen} sx={{ p: 0 }}>
+						<MenuIcon color='primary' fontSize='large' />
+					</IconButton>
+					<Menu
+						anchorEl={menuAnchorElem}
+						anchorOrigin={{
+							vertical: 'bottom',
+							horizontal: 'left',
+						}}
+						open={Boolean(menuAnchorElem)}
+						onClose={handleMenuClosed}
+					>
+						{menuItems(auth.isLoggedIn ? LEFT_PAGES : PAGES)}
+					</Menu>
+				</Box>
+				<Link href={routes.home} sx={{ mx: 'auto' }}>
+					<img
+						src={theme.palette.mode === 'dark' ? logoDark : logo}
+						alt='JeszCoChcesz'
+						style={{ display: 'block', height: '6rem', paddingBlock: '1rem' }}
+					/>
+				</Link>
+				<Box sx={{ flexGrow: 1, ml: 8, display: { xs: 'none', md: 'block' } }}>{LEFT_PAGES.map(pageToButton)}</Box>
+				{!auth.isLoggedIn && <Box sx={{ display: { xs: 'none', md: 'block' } }}>{RIGHT_PAGES.map(pageToButton)}</Box>}
+				<Tooltip title={`tryb ${theme.palette.mode === 'dark' ? 'jasny' : 'ciemny'}`} placement='top'>
+					<IconButton
+						sx={{ ml: 1, color: theme.palette.primary.main }}
+						onClick={colorMode.toggleColorMode}
+						color='inherit'
+					>
+						{theme.palette.mode === 'dark' ? <Brightness7Icon /> : <Brightness4Icon />}
+					</IconButton>
+				</Tooltip>
+				{auth.isLoggedIn && (
+					<Box sx={{ display: 'block' }}>
+						<IconButton onClick={handleProfileMenuOpen} sx={{ p: 2 }} title='Profil'>
+							<PersonIcon sx={{ fontSize: '3.5rem' }} color='secondary' />
+						</IconButton>
+						<Menu
+							anchorEl={profileMenuAnchorElem}
+							anchorOrigin={{
+								vertical: 'bottom',
+								horizontal: 'left',
+							}}
+							open={Boolean(profileMenuAnchorElem)}
+							onClose={handleProfileMenuClosed}
+						>
+							{[
+								auth.userRole === Role.User ? menuItems(USER_PAGES) : menuItems(PARTNER_PAGES),
+								<Divider key='divider' />,
+								<MenuItem key='logout' onClick={handleLogout}>
+									Wyloguj się
+								</MenuItem>,
+							]}
+						</Menu>
+					</Box>
+				)}
+
+				<Tooltip
+					title={cart.length === 0 ? <Typography fontSize='1.5rem'>Twój koszyk jest pusty</Typography> : ''}
+					placement='bottom-end'
+				>
+					<Button variant='contained' color='secondary' sx={{ borderRadius: '50%', ml: 2, p: 2 }}>
```

Comment:
Wydaje mi się, że tego buttona nie powinno tu być:
- nie ma żadnego `onClick`a -> skoro nic nie robi przy interakcji to nie musi być przyciskiem (i psuje accessibility)
- zbędnie pozwala klikać koszyk też kiedy nie ma żadnych dań

Jeśli chodziło tylko o kolor tła to IMO `Box` jak wcześniej wystarczy.